### PR TITLE
Update osquery elastic managed datastream naming convention

### PR DESCRIPTION
**Pull request by aleksmaus**
## What does this PR do?
Updated datastream naming convention for "osquery elastic managed" integration.
The final datastream default name is 
```
logs-osquery_managed.result-default
```
since 
```
logs-osquery.result-default
```
is taken  by the osquery filebeat based integration.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.

## Screenshots
Screen Shot 2021-03-08 at 9 36 29 PM

### CODE REVIEW

File: packages/osquery_elastic_managed/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,10 @@
 # newer versions go on top
+- version: "0.1.4"
+  changes:
+    - description: Updated datastream naming convention for "osquery elastic managed" intergration to logs-osquery_managed.result-default.
+      type: breaking change
```
Comment (ycombinator) [inline packages/osquery_elastic_managed/changelog.yml:5]: So it passes validation.

```suggestion
      type: breaking-change
```

File: packages/osquery_elastic_managed/changelog.yml [hunk 2]
```diff
@@ -1,4 +1,10 @@
 # newer versions go on top
+- version: "0.1.4"
```
Comment (ycombinator) [inline packages/osquery_elastic_managed/changelog.yml:2]: Since we're introducing a breaking change in this PR should the major version be bumped up? Or is it okay to go with what you have here since the package is still an `experimental` release. I also noticed that no version of this package has made it to the `production` package registry yet.

Comment (aleksmaus) [reply packages/osquery_elastic_managed/changelog.yml:2]: this is a new package still under development, haven't released yet, so not worrying about versioning as much yet, we are in process of fleshing out the details.

**Review by ycombinator — APPROVED** [review]
LGTM.

**Comment by ruflin** [comment]
Are the mappings of the two data streams identical? I'm good with the change but what I wondered is if maybe from a user perspective it does not matter in the end if it is collected by a managed or not managed osquery as long as the data is here. Having 2 packages with the same data stream is not really something we support yet but I wonder if in this scenario it would make sense?

In the future it should be possible to merge to two packages into 1 package and just have 2 integrations. At this point having just 1 data stream becomes more obvious.

So in summary, I would stick with what you have right now but maybe change it later on.
